# /ns/dyn_handler:
/**:
    ros__parameters:
    # 通信機器の設定
        device_name: /dev/ttyUSB0 # 通信するデバイス名
        baudrate: 2000000 # 通信速度
        latency_timer: 4 # 通信のインターバル
    # 通信の設定
        dyn_comm/retry_num: 10
        dyn_comm/inerval_msec: 5
        dyn_comm/varbose: false
    # サーボの設定
        init/expected_servo_num: 0
        init/auto_search_min_id: 0
        init/auto_search_max_id: 30
        init/auto_search_retry_times: 10
        init/hardware_error_auto_clean: true # 初期化時に Hardware error を自動でクリアするかどうか
        init/torque_auto_enable: true # 初期化時に Torque を自動でONにするかどうか
        term/torque_auto_disable: true # 終了時に Torque を自動でOFFにするかどうか
        middle/no_response_id_auto_remove_count: 0 # 通信が途切れた場合に何カウントで自動で削除するかどうか
    # ループの設定
        loop_rate: 50 # メインループの周期
        varbose_ratio: 300 # メインループの処理時間，通信の成功率を出力, ex 100なら100回に1回出力
        pub_ratio/present: # present_XXXを読み取る割合
            pwm:                 0 # この回数に一回present_pwmを読み取る
            current:             1 # この回数に一回present_currentを読み取る
            velocity:            1 # この回数に一回present_velocityを読み取る
            position:            1 # この回数に一回present_positionを読み取る
            velocity_trajectory: 0 # この回数に一回velocity_trajectoryを読み取る
            position_trajectory: 0 # この回数に一回position_trajectoryを読み取る
            input_voltage:      10 # この回数に一回present_input_voltageを読み取る
            temperature:        10 # この回数に一回present_temperatureを読み取る
        pub_ratio/status: 50 # この回数に一回 Status を読み取る, 0=初回のみ
        pub_ratio/goal: 10  # この回数に一回 Goal を読み取る, 0=初回のみ
        pub_ratio/gain: 300 # この回数に一回 Gain を読み取る, 0=初回のみ
        pub_ratio/limit: 300 # この回数に一回 Limit を読み取る, 0=初回のみ
        pub_ratio/error: 10 # この回数に一回 Hardware error を読み取る, 0=初回のみ
    # Read/Write方式
        use/fast_read: false
        use/split_read: false
        use/split_write: true
    # デバッグ用
        max_log_width: 6
        varbose/callback: false
        varbose/write_goal: true
        varbose/write_gain: false
        varbose/write_limit: false
        varbose/read_status/raw: false
        varbose/read_status/err: false
        varbose/read_present/raw: false
        varbose/read_present/err: false
        varbose/read_goal/raw: false
        varbose/read_goal/err: false
        varbose/read_gain/raw: false
        varbose/read_gain/err: false
        varbose/read_limit/raw: false
        varbose/read_limit/err: false
        varbose/read_hardware_error: false
